### PR TITLE
Remove mcid overriding

### PR DESCRIPTION
**Pull request by GribovskyPavel**

### CODE REVIEW

**Comment by GribovskyPavel** [comment]
[Paragraph is shown differently from Acrobat](https://trello.com/c/NZ8iM9SX/475-paragraph-is-shown-differently-from-acrobat)

File: src/core/bounding_boxes.js [hunk 1]
```diff
@@ -410,6 +412,22 @@ var BoundingBoxesCalculator = (function PartialEvaluatorClosure() {
         this.boundingBoxesStack.inc();
       }
 
+      // Prevent parsing of Marked content with same MCIDs
+      if (fn === OPS.beginMarkedContentProps && isDict(args[1]) && args[1].has('MCID')) {
```
Comment (KateOrient) [inline src/core/bounding_boxes.js]: Move this to `case OPS.beginMarkedContentProps`

File: src/core/bounding_boxes.js [hunk 2]
```diff
@@ -410,6 +412,22 @@ var BoundingBoxesCalculator = (function PartialEvaluatorClosure() {
...
+        const mcid = args[1].get('MCID');
+        if (this.mcidArray.includes(mcid)) {
+          this.sameMcidDepth++;
+          return;
+        } else {
+          this.mcidArray.push(mcid);
+        }
+      }
+
+      if (fn === OPS.endMarkedContent && this.sameMcidDepth !== 0) {
```
Comment (KateOrient) [inline src/core/bounding_boxes.js]: Move this to `case OPS.endMarkedContent`

**Review by KateOrient — CHANGES_REQUESTED** [review]

File: src/core/bounding_boxes.js [hunk 3]
```diff
@@ -540,7 +542,16 @@ var BoundingBoxesCalculator = (function PartialEvaluatorClosure() {
           break;
         case OPS.beginMarkedContentProps:
           if (isDict(args[1]) && args[1].has('MCID')) {
-            this.boundingBoxesStack.begin(args[1].get('MCID'));
+            const mcid = args[1].get('MCID');
+
+            // Prevent parsing of Marked content with same MCIDs
+            if (this.mcidArray.includes(mcid)) {
+              this.sameMcidDepth++;
+              return;
```
Comment (KateOrient) [inline src/core/bounding_boxes.js]: Use break instead of return as we don't have any logic after switch

File: src/core/bounding_boxes.js [hunk 4]
```diff
@@ -554,6 +565,11 @@ var BoundingBoxesCalculator = (function PartialEvaluatorClosure() {
           }
           break;
         case OPS.endMarkedContent:
+          if (this.sameMcidDepth !== 0) {
+            this.sameMcidDepth--;
+            return;
```
Comment (KateOrient) [inline src/core/bounding_boxes.js]: Use break instead of return as we don't have any logic after switch

**Review by KateOrient — CHANGES_REQUESTED** [review]

**Review by KateOrient — APPROVED** [review]

**Pull request merged**